### PR TITLE
Add-item-to-cart-187570882

### PR DESCRIPTION
**Pull request by rodriguecyber**
### What does this PR do?
 Adds item to cart by user 

### Description of Task to be completed?

he backend needs to handle the addition of items to the cart, including storing the cart in the buyer's account or session and calculating the new cart total.

The cart can be either be a new cart or the existing cart

### How should this be manually tested?

1.  run `yarn install` to install dependencies
2. run `yarn devstart` to build and run server
3.  Login with verified buyer account
4.  insert **Login Token** in header as Bearer
5.  send `POST`request  To this **END-POINT** `http://localhost:3000/api/cart/add`
6. **responses**:
 **200**: item adde in cart
**401:** you arevnot loged in or you are not a buyer
**500**: internal  server error
7. send `GET` request on `http://localhost:3000/api/cart` to view items in your Cart
 Pivot_tracker story: https://www.pivotaltracker.com/story/show/187570882

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ import { usersRouter } from './routes/user.route.js';
 import { productRouter } from './routes/productRoutes.js';
 import { sellerRouter } from './routes/sellerRoutes.js';
 import { rolesRouter } from './routes/roleRoutes.js';
+import cartRooutes from './routes/cartRoutes.js';
+import cartRoutes from './routes/cartRoutes.js';
```
Comment (kirengamartial) [inline src/server.ts:12]: you have the same imports in the same file and there is no where you are using it in that file if you are not using it you may remove it

**Review by kirengamartial — CHANGES_REQUESTED** [review]
see the little change i asked to make and see if you can change it

**Review by bahati10 — COMMENTED** [review]
Ensure to add newline at the end of each file
on the files you didn't add.

File: src/server.ts (hunk 1 above)
Comment (rodriguecyber) [reply src/server.ts:12]: corrected

**Review by Balinda21 — APPROVED** [review]
Good job @rodriguecyber you may consider those changes requested

**Review by Ishimwe7 — APPROVED** [review]
Good Job @rodriguecyber , everything is clear

**Review by kirengamartial — CHANGES_REQUESTED** [review]
look the changes i told to make and change them

File: src/models/cartModel.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    sequelize,
+    modelName: "carts",
+  },
+);
+
+Cart.belongsTo(User, { foreignKey: "userId", as: "user" });
+User.hasOne(Cart, { foreignKey: "userId", as: "cart" });
+Cart.hasMany(CartItem, { foreignKey: "cartId", as: "items" });
+
+export default Cart;
```
Comment (kirengamartial) [inline src/models/cartModel.ts]: add a new line for prettier

**Review by kirengamartial — CHANGES_REQUESTED** [review]
look for change too

**Review by kirengamartial — APPROVED** [review]
good job @rodriguecyber but see if you can solve that issue of GitGuardian

**Comment by rodriguecyber** [comment]
> good job @rodriguecyber but see if you can solve that issue of GitGuardian

that wasfor previous commit and i removed the file

**Review by DeliceLydia — APPROVED** [review]

**Pull request merged**